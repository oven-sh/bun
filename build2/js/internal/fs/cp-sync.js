(function (){"use strict";// build2/tmp/internal/fs/cp-sync.ts
var areIdentical = function(srcStat, destStat) {
  return destStat.ino && destStat.dev && destStat.ino === srcStat.ino && destStat.dev === srcStat.dev;
};
var isSrcSubdir = function(src, dest) {
  const srcArr = normalizePathToArray(src);
  const destArr = normalizePathToArray(dest);
  return ArrayPrototypeEvery.@call(srcArr, (cur, i) => destArr[i] === cur);
};
var cpSyncFn = function(src, dest, opts) {
  const { srcStat, destStat, skipped } = checkPathsSync(src, dest, opts);
  if (skipped)
    return;
  checkParentPathsSync(src, srcStat, dest);
  return checkParentDir(destStat, src, dest, opts);
};
var checkPathsSync = function(src, dest, opts) {
  if (opts.filter) {
    const shouldCopy = opts.filter(src, dest);
    if (isPromise(shouldCopy)) {
      throw new Error("Expected a boolean from the filter function, but got a promise. Use `fs.promises.cp` instead.");
    }
    if (!shouldCopy)
      return { __proto__: null, skipped: true };
  }
  const { srcStat, destStat } = getStatsSync(src, dest, opts);
  if (destStat) {
    if (areIdentical(srcStat, destStat)) {
      throw new Error("src and dest cannot be the same");
    }
    if (srcStat.isDirectory() && !destStat.isDirectory()) {
      throw new Error(`cannot overwrite directory ${src} with non-directory ${dest}`);
    }
    if (!srcStat.isDirectory() && destStat.isDirectory()) {
      throw new Error(`cannot overwrite non-directory ${src} with directory ${dest}`);
    }
  }
  if (srcStat.isDirectory() && isSrcSubdir(src, dest)) {
    throw new Error(`cannot copy ${src} to a subdirectory of self ${dest}`);
  }
  return { __proto__: null, srcStat, destStat, skipped: false };
};
var getStatsSync = function(src, dest, opts) {
  let destStat;
  const statFunc = opts.dereference ? (file) => statSync(file, { bigint: true }) : (file) => lstatSync(file, { bigint: true });
  const srcStat = statFunc(src);
  try {
    destStat = statFunc(dest);
  } catch (err) {
    if (err.code === "ENOENT")
      return { srcStat, destStat: null };
    throw err;
  }
  return { srcStat, destStat };
};
var checkParentPathsSync = function(src, srcStat, dest) {
  const srcParent = resolve(dirname(src));
  const destParent = resolve(dirname(dest));
  if (destParent === srcParent || destParent === parse(destParent).root)
    return;
  let destStat;
  try {
    destStat = statSync(destParent, { bigint: true });
  } catch (err) {
    if (err.code === "ENOENT")
      return;
    throw err;
  }
  if (areIdentical(srcStat, destStat)) {
    throw new Error(`cannot copy ${src} to a subdirectory of self ${dest}`);
  }
  return checkParentPathsSync(src, srcStat, destParent);
};
var checkParentDir = function(destStat, src, dest, opts) {
  const destParent = dirname(dest);
  if (!existsSync(destParent))
    mkdirSync(destParent, { recursive: true });
  return getStats(destStat, src, dest, opts);
};
var getStats = function(destStat, src, dest, opts) {
  const statSyncFn = opts.dereference ? statSync : lstatSync;
  const srcStat = statSyncFn(src);
  if (srcStat.isDirectory() && opts.recursive) {
    return onDir(srcStat, destStat, src, dest, opts);
  } else if (srcStat.isDirectory()) {
    throw new Error(`${src} is a directory (not copied)`);
  } else if (srcStat.isFile() || srcStat.isCharacterDevice() || srcStat.isBlockDevice()) {
    return onFile(srcStat, destStat, src, dest, opts);
  } else if (srcStat.isSymbolicLink()) {
    return onLink(destStat, src, dest, opts);
  } else if (srcStat.isSocket()) {
    throw new Error(`cannot copy a socket file: ${dest}`);
  } else if (srcStat.isFIFO()) {
    throw new Error(`cannot copy a FIFO pipe: ${dest}`);
  }
  throw new Error(`cannot copy an unknown file type: ${dest}`);
};
var onFile = function(srcStat, destStat, src, dest, opts) {
  if (!destStat)
    return copyFile(srcStat, src, dest, opts);
  return mayCopyFile(srcStat, src, dest, opts);
};
var mayCopyFile = function(srcStat, src, dest, opts) {
  if (opts.force) {
    unlinkSync(dest);
    return copyFile(srcStat, src, dest, opts);
  } else if (opts.errorOnExist) {
    throw new Error(`${dest} already exists`);
  }
};
var copyFile = function(srcStat, src, dest, opts) {
  copyFileSync(src, dest, opts.mode);
  if (opts.preserveTimestamps)
    handleTimestamps(srcStat.mode, src, dest);
  return setDestMode(dest, srcStat.mode);
};
var handleTimestamps = function(srcMode, src, dest) {
  if (fileIsNotWritable(srcMode))
    makeFileWritable(dest, srcMode);
  return setDestTimestamps(src, dest);
};
var fileIsNotWritable = function(srcMode) {
  return (srcMode & 128) === 0;
};
var makeFileWritable = function(dest, srcMode) {
  return setDestMode(dest, srcMode | 128);
};
var setDestMode = function(dest, srcMode) {
  return chmodSync(dest, srcMode);
};
var setDestTimestamps = function(src, dest) {
  const updatedSrcStat = statSync(src);
  return utimesSync(dest, updatedSrcStat.atime, updatedSrcStat.mtime);
};
var onDir = function(srcStat, destStat, src, dest, opts) {
  if (!destStat)
    return mkDirAndCopy(srcStat.mode, src, dest, opts);
  return copyDir(src, dest, opts);
};
var mkDirAndCopy = function(srcMode, src, dest, opts) {
  mkdirSync(dest);
  copyDir(src, dest, opts);
  return setDestMode(dest, srcMode);
};
var copyDir = function(src, dest, opts) {
  for (const dirent of readdirSync(src, { withFileTypes: true })) {
    const { name } = dirent;
    const srcItem = join(src, name);
    const destItem = join(dest, name);
    const { destStat, skipped } = checkPathsSync(srcItem, destItem, opts);
    if (!skipped)
      getStats(destStat, srcItem, destItem, opts);
  }
};
var onLink = function(destStat, src, dest, opts) {
  let resolvedSrc = readlinkSync(src);
  if (!opts.verbatimSymlinks && !isAbsolute(resolvedSrc)) {
    resolvedSrc = resolve(dirname(src), resolvedSrc);
  }
  if (!destStat) {
    return symlinkSync(resolvedSrc, dest);
  }
  let resolvedDest;
  try {
    resolvedDest = readlinkSync(dest);
  } catch (err) {
    if (err.code === "EINVAL" || err.code === "UNKNOWN") {
      return symlinkSync(resolvedSrc, dest);
    }
    throw err;
  }
  if (!isAbsolute(resolvedDest)) {
    resolvedDest = resolve(dirname(dest), resolvedDest);
  }
  if (isSrcSubdir(resolvedSrc, resolvedDest)) {
    throw new Error(`cannot copy ${resolvedSrc} to a subdirectory of self ${resolvedDest}`);
  }
  if (statSync(dest).isDirectory() && isSrcSubdir(resolvedDest, resolvedSrc)) {
    throw new Error(`cannot overwrite ${resolvedDest} with ${resolvedSrc}`);
  }
  return copyLink(resolvedSrc, dest);
};
var copyLink = function(resolvedSrc, dest) {
  unlinkSync(dest);
  return symlinkSync(resolvedSrc, dest);
};
var ArrayPrototypeEvery = @Array.prototype.every;
var ArrayPrototypeFilter = @Array.prototype.filter;
var StringPrototypeSplit = @String.prototype.split;
var normalizePathToArray = (path) => ArrayPrototypeFilter.@call(StringPrototypeSplit.@call(resolve(path), sep), Boolean);
var {
  chmodSync,
  copyFileSync,
  existsSync,
  lstatSync,
  mkdirSync,
  readdirSync,
  readlinkSync,
  statSync,
  symlinkSync,
  unlinkSync,
  utimesSync
} = @getInternalField(@internalModuleRegistry, 21) || @createInternalModuleById(21);
var { dirname, isAbsolute, join, parse, resolve, sep } = @getInternalField(@internalModuleRegistry, 30) || @createInternalModuleById(30);
var { isPromise } = @requireNativeModule("util/types");
return cpSyncFn})
