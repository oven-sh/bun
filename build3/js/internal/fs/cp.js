(function (){"use strict";// build3/tmp/internal/fs/cp.ts
async function cpFn(src, dest, opts) {
  const stats = await checkPaths(src, dest, opts);
  const { srcStat, destStat, skipped } = stats;
  if (skipped)
    return;
  await checkParentPaths(src, srcStat, dest);
  return checkParentDir(destStat, src, dest, opts);
}
async function checkPaths(src, dest, opts) {
  if (opts.filter && !await opts.filter(src, dest)) {
    return { __proto__: null, skipped: true };
  }
  const { 0: srcStat, 1: destStat } = await getStats(src, dest, opts);
  if (destStat) {
    if (areIdentical(srcStat, destStat)) {
      throw new Error("Source and destination must not be the same.");
    }
    if (srcStat.isDirectory() && !destStat.isDirectory()) {
      throw new Error(`cannot overwrite directory ${src} with non-directory ${dest}`);
    }
    if (!srcStat.isDirectory() && destStat.isDirectory()) {
      throw new Error(`cannot overwrite non-directory ${src} with directory ${dest}`);
    }
  }
  if (srcStat.isDirectory() && isSrcSubdir(src, dest)) {
    throw new Error(`cannot copy ${src} to a subdirectory of self ${dest}`);
  }
  return { __proto__: null, srcStat, destStat, skipped: false };
}
var areIdentical = function(srcStat, destStat) {
  return destStat.ino && destStat.dev && destStat.ino === srcStat.ino && destStat.dev === srcStat.dev;
};
var getStats = function(src, dest, opts) {
  const statFunc = opts.dereference ? (file) => stat(file, { bigint: true }) : (file) => lstat(file, { bigint: true });
  return SafePromiseAll([
    statFunc(src),
    PromisePrototypeThen.@call(statFunc(dest), @undefined, (err) => {
      if (err.code === "ENOENT")
        return null;
      throw err;
    })
  ]);
};
async function checkParentDir(destStat, src, dest, opts) {
  const destParent = dirname(dest);
  const dirExists = await pathExists(destParent);
  if (dirExists)
    return getStatsForCopy(destStat, src, dest, opts);
  await mkdir(destParent, { recursive: true });
  return getStatsForCopy(destStat, src, dest, opts);
}
var pathExists = function(dest) {
  return PromisePrototypeThen(stat(dest), () => true, (err) => err.code === "ENOENT" ? false : PromiseReject(err));
};
async function checkParentPaths(src, srcStat, dest) {
  const srcParent = resolve(dirname(src));
  const destParent = resolve(dirname(dest));
  if (destParent === srcParent || destParent === parse(destParent).root) {
    return;
  }
  let destStat;
  try {
    destStat = await stat(destParent, { bigint: true });
  } catch (err) {
    if (err.code === "ENOENT")
      return;
    throw err;
  }
  if (areIdentical(srcStat, destStat)) {
    throw new Error(`cannot copy ${src} to a subdirectory of self ${dest}`);
  }
  return checkParentPaths(src, srcStat, destParent);
}
var isSrcSubdir = function(src, dest) {
  const srcArr = normalizePathToArray(src);
  const destArr = normalizePathToArray(dest);
  return ArrayPrototypeEvery.@call(srcArr, (cur, i) => destArr[i] === cur);
};
async function getStatsForCopy(destStat, src, dest, opts) {
  const statFn = opts.dereference ? stat : lstat;
  const srcStat = await statFn(src);
  if (srcStat.isDirectory() && opts.recursive) {
    return onDir(srcStat, destStat, src, dest, opts);
  } else if (srcStat.isDirectory()) {
    throw new Error(`${src} is a directory (not copied)`);
  } else if (srcStat.isFile() || srcStat.isCharacterDevice() || srcStat.isBlockDevice()) {
    return onFile(srcStat, destStat, src, dest, opts);
  } else if (srcStat.isSymbolicLink()) {
    return onLink(destStat, src, dest, opts);
  } else if (srcStat.isSocket()) {
    throw new Error(`cannot copy a socket file: ${dest}`);
  } else if (srcStat.isFIFO()) {
    throw new Error(`cannot copy a FIFO pipe: ${dest}`);
  }
  throw new Error(`cannot copy an unknown file type: ${dest}`);
}
var onFile = function(srcStat, destStat, src, dest, opts) {
  if (!destStat)
    return _copyFile(srcStat, src, dest, opts);
  return mayCopyFile(srcStat, src, dest, opts);
};
async function mayCopyFile(srcStat, src, dest, opts) {
  if (opts.force) {
    await unlink(dest);
    return _copyFile(srcStat, src, dest, opts);
  } else if (opts.errorOnExist) {
    throw new Error(`${dest} already exists`);
  }
}
async function _copyFile(srcStat, src, dest, opts) {
  await copyFile(src, dest, opts.mode);
  if (opts.preserveTimestamps) {
    return handleTimestampsAndMode(srcStat.mode, src, dest);
  }
  return setDestMode(dest, srcStat.mode);
}
async function handleTimestampsAndMode(srcMode, src, dest) {
  if (fileIsNotWritable(srcMode)) {
    await makeFileWritable(dest, srcMode);
    return setDestTimestampsAndMode(srcMode, src, dest);
  }
  return setDestTimestampsAndMode(srcMode, src, dest);
}
var fileIsNotWritable = function(srcMode) {
  return (srcMode & 128) === 0;
};
var makeFileWritable = function(dest, srcMode) {
  return setDestMode(dest, srcMode | 128);
};
async function setDestTimestampsAndMode(srcMode, src, dest) {
  await setDestTimestamps(src, dest);
  return setDestMode(dest, srcMode);
}
var setDestMode = function(dest, srcMode) {
  return chmod(dest, srcMode);
};
async function setDestTimestamps(src, dest) {
  const updatedSrcStat = await stat(src);
  return utimes(dest, updatedSrcStat.atime, updatedSrcStat.mtime);
}
var onDir = function(srcStat, destStat, src, dest, opts) {
  if (!destStat)
    return mkDirAndCopy(srcStat.mode, src, dest, opts);
  return copyDir(src, dest, opts);
};
async function mkDirAndCopy(srcMode, src, dest, opts) {
  await mkdir(dest);
  await copyDir(src, dest, opts);
  return setDestMode(dest, srcMode);
}
async function copyDir(src, dest, opts) {
  const dir = await opendir(src);
  for await (const { name } of dir) {
    const srcItem = join(src, name);
    const destItem = join(dest, name);
    const { destStat, skipped } = await checkPaths(srcItem, destItem, opts);
    if (!skipped)
      await getStatsForCopy(destStat, srcItem, destItem, opts);
  }
}
async function onLink(destStat, src, dest, opts) {
  let resolvedSrc = await readlink(src);
  if (!opts.verbatimSymlinks && !isAbsolute(resolvedSrc)) {
    resolvedSrc = resolve(dirname(src), resolvedSrc);
  }
  if (!destStat) {
    return symlink(resolvedSrc, dest);
  }
  let resolvedDest;
  try {
    resolvedDest = await readlink(dest);
  } catch (err) {
    if (err.code === "EINVAL" || err.code === "UNKNOWN") {
      return symlink(resolvedSrc, dest);
    }
    throw err;
  }
  if (!isAbsolute(resolvedDest)) {
    resolvedDest = resolve(dirname(dest), resolvedDest);
  }
  if (isSrcSubdir(resolvedSrc, resolvedDest)) {
    throw new Error(`cannot copy ${resolvedSrc} to a subdirectory of self ${resolvedDest}`);
  }
  const srcStat = await stat(src);
  if (srcStat.isDirectory() && isSrcSubdir(resolvedDest, resolvedSrc)) {
    throw new Error(`cannot overwrite ${resolvedDest} with ${resolvedSrc}`);
  }
  return copyLink(resolvedSrc, dest);
}
async function copyLink(resolvedSrc, dest) {
  await unlink(dest);
  return symlink(resolvedSrc, dest);
}
var { chmod, copyFile, lstat, mkdir, opendir, readlink, stat, symlink, unlink, utimes } = @getInternalField(@internalModuleRegistry, 22) || @createInternalModuleById(22);
var { dirname, isAbsolute, join, parse, resolve, sep } = @getInternalField(@internalModuleRegistry, 30) || @createInternalModuleById(30);
var SafePromiseAll = @Promise.all;
var PromisePrototypeThen = @Promise.prototype.then;
var PromiseReject = @Promise.reject;
var ArrayPrototypeFilter = @Array.prototype.filter;
var StringPrototypeSplit = @String.prototype.split;
var ArrayPrototypeEvery = @Array.prototype.every;
var normalizePathToArray = (path) => ArrayPrototypeFilter.@call(StringPrototypeSplit(resolve(path), sep), Boolean);
return cpFn})
