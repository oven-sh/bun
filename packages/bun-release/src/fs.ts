import path from "path";
import fs from "fs";
import os from "os";
import crypto from "crypto";
import { debug } from "./console";

export function join(...paths: (string | string[])[]): string {
  return path.join(...paths.flat(2));
}

export function basename(...paths: (string | string[])[]): string {
  return path.basename(join(...paths));
}

export function tmp(): string {
  const tmpdir = process.env["RUNNER_TEMP"] ?? os.tmpdir();
  const dir = fs.mkdtempSync(join(tmpdir, "bun-"));
  debug("tmp", dir);
  return dir;
}

export function rm(path: string): void {
  debug("rm", path);
  try {
    fs.rmSync(path, { recursive: true });
    return;
  } catch (error) {
    debug("fs.rmSync failed", error);
    // Did not exist before Node.js v14.
    // Attempt again with older, slower implementation.
  }
  let stats: fs.Stats;
  try {
    stats = fs.lstatSync(path);
  } catch (error) {
    debug("fs.lstatSync failed", error);
    // The file was likely deleted, so return early.
    return;
  }
  if (!stats.isDirectory()) {
    fs.unlinkSync(path);
    return;
  }
  try {
    fs.rmdirSync(path, { recursive: true });
    return;
  } catch (error) {
    debug("fs.rmdirSync failed", error);
    // Recursive flag did not exist before Node.js X.
    // Attempt again with older, slower implementation.
  }
  for (const filename of fs.readdirSync(path)) {
    rm(join(path, filename));
  }
  fs.rmdirSync(path);
}

export function rename(path: string, newPath: string): void {
  debug("rename", path, newPath);
  try {
    fs.renameSync(path, newPath);
    return;
  } catch (error) {
    debug("fs.renameSync failed", error);
    // If there is an error, delete the new path and try again.
  }
  try {
    rm(newPath);
  } catch (error) {
    debug("rm failed", error);
    // The path could have been deleted already.
  }
  fs.renameSync(path, newPath);
}

export function write(dst: string, content: string | ArrayBuffer | ArrayBufferView): void {
  debug("write", dst);
  try {
    fs.writeFileSync(dst, content);
    return;
  } catch (error) {
    debug("fs.writeFileSync failed", error);
    // If there is an error, ensure the parent directory
    // exists and try again.
    try {
      fs.mkdirSync(path.dirname(dst), { recursive: true });
    } catch (error) {
      debug("fs.mkdirSync failed", error);
      // The directory could have been created already.
    }
    fs.writeFileSync(dst, content);
  }
}

export function writeJson(path: string, json: object, force?: boolean): void {
  let value = json;
  if (!force && exists(path)) {
    try {
      const existing = JSON.parse(read(path));
      value = {
        ...existing,
        ...json,
      };
    } catch {
      value = json;
    }
  }
  write(path, `${JSON.stringify(value, undefined, 2)}\n`);
}

export function read(path: string): string {
  debug("read", path);
  return fs.readFileSync(path, "utf-8");
}

export function blob(path: string): Blob {
  debug("blob", path);
  if ("Bun" in globalThis) {
    return Bun.file(path);
  }
  const buffer = fs.readFileSync(path);
  return new Blob([buffer], {
    type: path.endsWith(".zip") ? "application/zip" : path.endsWith(".txt") ? "text/plain" : "application/octet-stream",
  });
}

export function hash(content: string | crypto.BinaryLike): string {
  debug("hash", content);
  return crypto
    .createHash("sha256")
    .update(typeof content === "string" ? fs.readFileSync(content) : content)
    .digest("hex");
}

export function chmod(path: string, mode: fs.Mode): void {
  debug("chmod", path, mode);
  fs.chmodSync(path, mode);
}

export function copy(path: string, newPath: string): void {
  debug("copy", path, newPath);
  try {
    fs.copyFileSync(path, newPath);
    return;
  } catch (error) {
    debug("fs.copyFileSync failed", error);
  }
  write(newPath, read(path));
}

export function exists(path: string): boolean {
  debug("exists", path);
  try {
    return fs.existsSync(path);
  } catch (error) {
    debug("fs.existsSync failed", error);
  }
  return false;
}
