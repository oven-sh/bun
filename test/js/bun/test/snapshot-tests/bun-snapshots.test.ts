import fs from "fs";

beforeAll(() => {
  fs.rmSync(import.meta.dir + "/__snapshots__/bun-snapshots.test.ts.snap", { force: true });
});

test("it will create a snapshot file if it doesn't exist", () => {
  expect({ a: { b: { c: false } }, c: 2, jkfje: 99238 }).toMatchSnapshot({ a: { b: { c: expect.any(Boolean) } } });
  expect({ a: { b: { c: "string" } }, c: 2, jkfje: 99238 }).toMatchSnapshot({ a: { b: { c: expect.any(String) } } });
  expect({ a: { b: { c: 4 } }, c: 2, jkfje: 99238 }).toMatchSnapshot({ a: { b: { c: expect.any(Number) } } });
  expect({ a: { b: { c: 2n } }, c: 2, jkfje: 99238 }).toMatchSnapshot({ a: { b: { c: expect.any(BigInt) } } });
  expect({ a: new Date() }).toMatchSnapshot({ a: expect.any(Date) });
  expect({ j: 2, a: "any", b: "any2" }).toMatchSnapshot({ j: expect.any(Number), a: "any", b: expect.any(String) });
  expect({ j: /regex/, a: "any", b: "any2" }).toMatchSnapshot({
    j: expect.any(RegExp),
    a: "any",
    b: expect.any(String),
  });
});

describe("toMatchSnapshot errors", () => {
  it("should throw if property matchers exist and received is not an object", () => {
    expect(() => {
      expect(1).toMatchSnapshot({ a: 1 });
    }).toThrow();
  });
  it("should throw if property matchers don't match", () => {
    expect(() => {
      expect({ a: 3 }).toMatchSnapshot({ a: 1 });
    }).toThrow();
    expect(() => {
      expect({ a: 3 }).toMatchSnapshot({ a: expect.any(Date) });
    }).toThrow();
    expect(() => {
      expect({ a: 3 }).toMatchSnapshot({ a: expect.any(String) });
    }).toThrow();
    expect(() => {
      expect({ a: 4n }).toMatchSnapshot({ a: expect.any(Number) });
    }).toThrow();
    expect(() => {
      expect({ a: 3 }).toMatchSnapshot({ a: expect.any(BigInt) });
    }).toThrow();
  });
  it("should throw if arguments are in the wrong order", () => {
    expect(() => {
      expect({ a: "oops" }).toMatchSnapshot("wrong spot", { a: "oops" });
    }).toThrow();
    expect(() => {
      expect({ a: "oops" }).toMatchSnapshot({ a: "oops" }, "right spot");
    }).not.toThrow();
  });

  it("should throw if expect.any() doesn't received a constructor", () => {
    expect(() => {
      expect({ a: 4 }).toMatchSnapshot({ a: expect.any() });
    }).toThrow();
    expect(() => {
      expect({ a: 5 }).toMatchSnapshot({ a: expect.any(5) });
    }).toThrow();
    expect(() => {
      expect({ a: 4 }).toMatchSnapshot({ a: expect.any("not a constructor") });
    }).toThrow();
  });
});
