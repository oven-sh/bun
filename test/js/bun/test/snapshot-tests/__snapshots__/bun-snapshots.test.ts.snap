// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`toMatchSnapshot errors should throw if arguments are in the wrong order: right spot 1`] = `
{
  "a": "oops",
}
`;

exports[`it will create a snapshot file if it doesn't exist 1`] = `
{
  "a": {
    "b": {
      "c": Any<Boolean>,
    },
  },
  "c": 2,
  "jkfje": 99238,
}
`;

exports[`it will create a snapshot file if it doesn't exist 2`] = `
{
  "a": {
    "b": {
      "c": Any<String>,
    },
  },
  "c": 2,
  "jkfje": 99238,
}
`;

exports[`it will create a snapshot file if it doesn't exist 3`] = `
{
  "a": {
    "b": {
      "c": Any<Number>,
    },
  },
  "c": 2,
  "jkfje": 99238,
}
`;

exports[`it will create a snapshot file if it doesn't exist 4`] = `
{
  "a": {
    "b": {
      "c": Any<BigInt>,
    },
  },
  "c": 2,
  "jkfje": 99238,
}
`;

exports[`it will create a snapshot file if it doesn't exist 5`] = `
{
  "a": Any<Date>,
}
`;

exports[`it will create a snapshot file if it doesn't exist 6`] = `
{
  "a": "any",
  "b": Any<String>,
  "j": Any<Number>,
}
`;

exports[`it will create a snapshot file if it doesn't exist 7`] = `
{
  "a": "any",
  "b": Any<String>,
  "j": Any<RegExp>,
}
`;
